application/src/test/java/org/opentripplanner/apis/gtfs/mapping/routerequest/LegacyRouteRequestMapperTest.java

```diff
@@ -144,9 +145,36 @@ static Stream<Arguments> transportModesCases() {
     );
   }
 
+  static Stream<Arguments> transportModesCasesWithCoach() {
+    return Stream.of(
+      of(
+        List.of(mode("BUS")),
+        "[TransitFilterRequest{select: [SelectRequest{transportModes: [BUS]}]}]"
+      ),
+      of(
+        List.of(mode("BUS"), mode("COACH")),
+        "[TransitFilterRequest{select: [SelectRequest{transportModes: [BUS, COACH]}]}]"
+      ),
+      of(
+        List.of(mode("BUS"), mode("MONORAIL")),
+        "[TransitFilterRequest{select: [SelectRequest{transportModes: [BUS, MONORAIL]}]}]"
+      )
+    );
+  }
+
   @ParameterizedTest
   @MethodSource("transportModesCases")
   void modes(List<Map<String, Object>> modes, String expectedFilters) {
+    OTPFeature.GtfsCoach.testOff(() -> testModes(modes, expectedFilters));
+  }
+
+  @ParameterizedTest
+  @MethodSource("transportModesCasesWithCoach")
+  void modesWithCoach(List<Map<String, Object>> modes, String expectedFilters) {
+    OTPFeature.GtfsCoach.testOn(() -> testModes(modes, expectedFilters));
```

Comment:
Does this feature flag being on/off affect the results of this test at all?

Reply:
Yes. See the difference with the original test.

Reply:
I was under the impression that the ApiRequestMode was only used by the REST API but seems like it's used elsewhere as well.

Reply:
It is also used for the `plan` field in the GTFS GraphQL API as well. Therefore, without this PR, entering `COACH` into the query will make it error.